### ISSUE TITLE
Different bytecode Hardhat vs Sourcify

### ISSUE THREAD

**Issue by kuzdogan** [ethereum/sourcify#1065]
We were reported a case when the compiled bytecodes end up different when compiled with Hardhat and with Sourcify. I suspect this is a case of https://github.com/ethereum/solidity/issues/14250

[meter-2707667c8a3e7035bf25c34ad0a1ec81.json.txt](https://github.com/ethereum/sourcify/files/11768095/meter-2707667c8a3e7035bf25c34ad0a1ec81.json.txt)

Chain: Meter (82)
Address: 0x07FfD62571b586dCA7eA1DC71C10a6270e078F01
Contract: N2MERC721

Onchain vs recompiled runtime bytecodes:
[82-0x07FfD62571b586dCA7eA1DC71C10a6270e078F01-recompiled-deployed.txt](https://github.com/ethereum/sourcify/files/11768122/82-0x07FfD62571b586dCA7eA1DC71C10a6270e078F01-recompiled-deployed.txt)
[82-0x07FfD62571b586dCA7eA1DC71C10a6270e078F01-onchain-deployed.txt](https://github.com/ethereum/sourcify/files/11768123/82-0x07FfD62571b586dCA7eA1DC71C10a6270e078F01-onchain-deployed.txt)

They are different in many places:
```
git diff --word-diff --word-diff-regex=. 82-0x07FfD62571b586dCA7eA1DC71C10a6270e078F01-onchain-deployed.txt 82-0x07FfD62571b586dCA7eA1DC71C10a6270e078F01-recompiled-deployed.txt
```
![image](https://github.com/ethereum/sourcify/assets/13069972/4c066326-ea19-4cd2-a1c2-3f20434e9246)

**Comment by kuzdogan** [comment]
I can confirm this gets resolved with `useAllSources` i.e. using all other sources provided alongside the contract. 

However the bad news is this is not the https://github.com/ethereum/solidity/issues/14250 case because I was able to reproduce the issue in 0.8.21 whereas the above issue must have been fixed in 0.8.21.

This looks similar to https://github.com/ethereum/sourcify/issues/618 because the metadata hashes are identical but the bytecodes are different. However, that specific case must also have been fixed in 0.7.1. So I suspect this is yet another bug in Solidity

**Comment by kuzdogan** [comment]
Opened an issue in Solidity https://github.com/ethereum/solidity/issues/14494

**Comment by marcocastignoli** [comment]
I'm handling this cases in this way, if you also think is enough we can close this:

<img width="1031" alt="Screenshot 2023-11-22 alle 10 26 39" src="https://github.com/ethereum/sourcify/assets/10226180/c1d67ab5-d484-4efe-b92a-1ace5fcd8c5d">

**Comment by kuzdogan** [comment]
Does this handle the example case above? Does the test pass?

**Comment by marcocastignoli** [comment]
fixed with f1be6297bcf53041146691aef709c3635d87df97
